### PR TITLE
feat: impl region engine for mito

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Implement RegionEngine for mito. From now on all requests can be passed to the corresponding engine.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/store-api/src/metadata.rs [hunk 1]
```diff
@@ -45,6 +47,32 @@ pub struct ColumnMetadata {
     pub column_id: ColumnId,
 }
 
+impl ColumnMetadata {
+    /// Construct `Self` from protobuf struct [ColumnDef]
+    pub fn from_column_def(column_def: ColumnDef) -> Result<Self> {
```
Comment (evenyag) [inline src/store-api/src/metadata.rs:52]: nitpick: try_from
```suggestion
    pub fn try_from_column_def(column_def: ColumnDef) -> Result<Self> {
```

Comment (evenyag) [inline src/store-api/src/metadata.rs:52]: question(non-blocking): Should we implement `TryFrom`?

File: src/store-api/src/region_request.rs [hunk 2]
```diff
@@ -32,6 +34,85 @@ pub enum RegionRequest {
     Compact(RegionCompactRequest),
 }
 
+impl RegionRequest {
+    /// Convert [Body](region_request::Body) to a group of [RegionRequest] with region id.
+    /// Inserts/Deletes request might become multiple requests. Others are one-to-one.
+    // TODO: implement alter request
+    #[allow(unreachable_code)]
+    pub fn from_request_body(
```
Comment (evenyag) [inline src/store-api/src/region_request.rs:42]: This should be try_from
```suggestion
    pub fn try_from_request_body(
```

File: src/store-api/src/metadata.rs (hunk 1 above)
Comment (waynexia) [reply src/store-api/src/metadata.rs:52]: I prefer not to. It's hard to find

**Review by evenyag — APPROVED** [review]
...

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**